source/common/router/header_formatter.cc

```diff
@@ -19,6 +115,8 @@ RequestInfoHeaderFormatter::RequestInfoHeaderFormatter(const std::string& field_
     field_extractor_ = [](const Envoy::AccessLog::RequestInfo& request_info) {
       return request_info.getDownstreamAddress();
     };
+  } else if (StringUtil::startsWith(field_name.c_str(), "UPSTREAM_METADATA")) {
+    field_extractor_ = parseUpstreamMetadataField(field_name.substr(17));
```

Comment:
nit: Might use sizeof() here or some other constant to avoid the magic '17'

Reply:
+1